### ISSUE TITLE
support for min/max generation

### ISSUE THREAD

**Issue by abstratt** [abstratt/cloudfier#63]
Provide support for min/max aggregators.

The following test case:

```
    def void testMax() throws CoreException, IOException {
        var source = '''
        model crm;
            class Company
                attribute revenue : Double;
                query highestRevenue() : Double;
                begin
                    return Company extent.max((c : Company) : Double {
                        c.revenue
                    });
                end;
                
            end;
	    end.
        '''
        parseAndCheck(source)
        val op = getOperation('crm::Company::highestRevenue')
        val root = getStatementSourceAction(op)
        val generated = new QueryActionGenerator(repository).generateAction(root)
        AssertHelper.assertStringsEqual(
            '''
	            cq
	            	.distinct(true)
	            	.select(cb.max(rental_.get("revenue")))
            ''', generated.toString)
    }

```

yields:

```
org.junit.ComparisonFailure: expected:<[cq.distinct(true).select(cb.max(rental_.get("revenue")))]> but was:<[null/*Unsupported Collection operation:max*/]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.abstratt.mdd.core.tests.harness.AssertHelper.assertStringsEqual(AssertHelper.java:32)
	at com.abstratt.mdd.target.tests.jee.QueryActionGeneratorTests.testMax(QueryActionGeneratorTests.java:314)
```

**Comment by abstratt** [comment]
Fixed in 4e9b9ff3af946ed207188599eb1170ec59579fce.

**Comment by abstratt** [comment]
Fixed in 4e9b9ff3af946ed207188599eb1170ec59579fce.
